roles/adoption_osp_deploy/tasks/prepare_overcloud.yml

```diff
@@ -0,0 +1,171 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Prepare enviornment for 17.1 overcloud deployment
+  vars:
+    _network_data_file: >-
+      {{
+        [cifmw_adoption_source_scenario_path,
+         _adoption_source_scenario.overcloud.network_data_file
+        ] | path_join
+      }}
+    _network_data_file_name: "{{ _network_data_file | basename }}"
+    _network_data_file_dest: "{{ ansible_user_dir }}/{{ _network_data_file_name }}"
+    _vips_data_file: >-
+      {{
+        [cifmw_adoption_source_scenario_path,
+         _adoption_source_scenario.overcloud.vips_data_file
+        ] | path_join
+      }}
+    _vips_data_file_name: "{{ _vips_data_file | basename }}"
+    _vips_data_file_dest: "/home/zuul/{{ _vips_data_file_name }}"
+    _source_cmd: "source {{ ansible_user_dir }}/stackrc"
+  block:
+    - name: Ensure overcloud vms are started
+      community.libvirt.virt:
+        state: running
+        name: "cifmw-{{ _vm }}"
+        uri: "qemu:///system"
+      loop: "{{ _tripleo_nodes }}"
+      loop_control:
+        loop_var: _vm
+        pause: 1
+
+    - name: Ensure repos are setup in overcloud nodes
+      when: "'undercloud' not in _vm"
+      delegate_to: "{{ _vm }}"
+      vars:
+        osp_17_repos:
+          - rhel-9-for-x86_64-baseos-eus-rpms
+          - rhel-9-for-x86_64-appstream-eus-rpms
+          - rhel-9-for-x86_64-highavailability-eus-rpms
+          - openstack-17.1-for-rhel-9-x86_64-rpms
+          - fast-datapath-for-rhel-9-x86_64-rpms
+          - rhceph-6-tools-for-rhel-9-x86_64-rpms
+      become: true
+      ansible.builtin.command:
```

Comment:
Before running this command nodes must be registered RHSM. For manual run of the playbook we nodes could be registered using redhat_subscription module[1]
[1]https://docs.ansible.com/ansible/latest/collections/community/general/redhat_subscription_module.html

Reply:
I've added now a task to register nodes when credentials are passed